internal/server/data/data_test.go

```diff
@@ -213,3 +214,37 @@ func TestDB_Begin(t *testing.T) {
 		})
 	})
 }
+
+func TestLongRunningQueriesAreCancelled(t *testing.T) {
+	runDBTests(t, func(t *testing.T, db *DB) {
+
+		t.Run("Gorm", func(t *testing.T) {
+			started := time.Now()
+
+			ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
```

Comment:
Could use a timeout of maybe `200 * time.Millisecond` to spend less time waiting on the test?

Reply:
yeah, was a bit worried about slow CI servers